src/nvim/ex_getln.c

```diff
@@ -2393,10 +2394,11 @@ static int command_line_changed(CommandLineState *s)
       && !vpeekc_any()
       && cmdpreview_may_show(s)) {
     // 'inccommand' preview has been shown.
-  } else if (cmdpreview) {
-    cmdpreview = false;
-    update_screen(UPD_SOME_VALID);  // Clear 'inccommand' preview.
   } else {
+    cmdpreview = false;
+    if (prev_cmdpreview) {
+      update_screen(UPD_SOME_VALID);  // Clear 'inccommand' preview.
+    }
```

Comment:
I don't understand the reason for adding the `prev_cmdpreview` variable when we can just do

```suggestion
    if (cmdpreview) {
      cmdpreview = false;
      update_screen(UPD_SOME_VALID);  // Clear 'inccommand' preview.
    }
```

Reply:
That doesn't work if preview callback returns 0, because `cmdpreview` is already set.

Reply:
Ah right, forgot that `cmdpreview_may_show` can set `cmdpreview`